operator/pkg/watch/template_change.go

```diff
@@ -53,38 +54,50 @@ func (h *TemplateChangeHandler) Watch(ctx context.Context) handler.MapFunc {
 			Namespace: template.GetNamespace(),
 			Name:      template.GetName(),
 		}
-		for _, kyma := range kymas.Items {
-			globalChannelMatch := kyma.Spec.Channel == channel
-			requeueKyma := false
-
-			for _, component := range kyma.Spec.Components {
-				if component.Name == controller {
-					// check component level channel on matching component
-					requeueKyma = (component.Channel == "" && globalChannelMatch) ||
-						component.Channel == channel
-
-					if requeueKyma {
-						break
-					}
-				}
-			}
+		requests = handleRequest(ctx, kymas, channel, controller, templateNamespacedName, requests)
 
-			if !requeueKyma {
-				continue
-			}
+		return requests
+	}
+}
+
+func handleRequest(ctx context.Context, kymas *v1alpha1.KymaList, channel v1alpha1.Channel,
```

Comment:
What check caused this change? For me its not much easier to read this way tbh...

Reply:
pkg/watch/template_change.go:25: Function 'Watch' is too long (68 > 60) ([funlen](https://github.com/ultraware/funlen)), basically complains long method

Reply:
I actually just use IDE to extracted a method, tbh. Do you have any suggestions? maybe define several more concrete methods.

Reply:
what I usually do is use one function per loop, that makes things a bit easier, but the issue i have is with the amount of parameters it passes.

Reply:
extracted a piece of code as more readable method